### DISCUSSION TITLE
How to override the default fileMatch for Terraform manager?

### DISCUSSION THREAD

**Discussion by kayman-mk** [renovatebot/renovate#10845] [Request Help]
We are using Renovate to update our dependencies. Great tool! Unfortunately the Terraform manager updates the provider versions of modules too. We don't want this as we specify minimum versions.

I know how to exclude whole paths from the Renovate analysis (ignorePaths) and how to add further RexEx to analyze files not included in the default RexEx of the manager. But how to replace the default RegEx for fileMatch? I am looking for something like

`"terraform" {"override-default-fileMatch": "^(?!modules\/.*providers.tf).*\.tf$"}`

Any ideas how to do that?

**Comment by viceice** [comment]
```json
{
    "terraform": {
          "fileMatch": "^(?!modules\\/.*providers.tf).*\\.tf$"
     }
}
```

You need to double escape regex specials chars

**Reply by rarkins** [reply]
Should be an array.. also isn't it additive rather than replacing?

**Reply by viceice** [reply]
Didn't checked, just guessing. 🤷‍♂️😅

**Reply by kayman-mk** [reply]
Unfortunately it's additive, yes.

Thought about not checking all Terraform modules. But it's not an option as they might include other modules which version may change.

**Reply by rarkins** [reply]
You can add ignorePaths config inside terraform in config

**Reply by rarkins** [reply]
IgnorePaths will be evaluated before fileMatch and can be manager-specific so won't that achieve the same result for you?

**Reply by kayman-mk** [reply]
But ignorePath skips whole paths, right? I need to skip certain files only. I don't want the manager to analyze "provider.tf" which is part of the default fileMatch (*.tf). Or can I add my RegEx to the ignorePath?

**Reply by rarkins** [reply]
ignorePaths can be down to the file level if you need

**Comment by kayman-mk** [comment]
Added `ignorePaths: ['**/modules/**/provider.tf'],` to config.js which works like a charme.
